dias/tracker.py

```diff
@@ -0,0 +1,418 @@
+                analyzer_find_jump = Analyzer.create(
+                    name="find_jump", file_type_id=file_type_rows["chimecal"]
+                )
+                analyzer_flag_rfi = Analyzer.create(
+                    name="flag_rfi", file_type_id=file_type_rows["chimestack"]
+                )
+                analyzer_sensitivity = Analyzer.create(
+                    name="sensitivity", file_type_id=file_type_rows["chimestack"]
+                )
+                analyzer_source_spectra = Analyzer.create(
+                    name="source_spectra", file_type_id=file_type_rows["chimestack"]
+                )
+                analyzer_thermaldata = Analyzer.create(
+                    name="thermaldata", file_type_id=file_type_rows["chimetiming"]
+                )
+
+            if not Analyzer.get(Analyzer.name == self.dias_task_name):
+                print(
+                    "File Tracker ERROR: Provided analyzer is not registered in database"
+                )
+                return None
+
+            if not db.table_exists("file"):
+                print("File Tracker: Creating file table")
+                db.create_tables([FileTable])
+
+                files = []
+                # grab a list of files on disk of FILETYPE
+                for ft in FILETYPES:
+                    files.extend(
+                        glob.glob(
+                            os.path.join(
+                                self.base_path,
+                                f"*_{ft}_corr",
+                                "*.h5",
+                            )
+                        )
+                    )
+
+                self.insert_files(files)
+
+            if not db.table_exists("processed"):
+                print("File Tracker: Creating processed table")
+                db.create_tables([Processed])
+
+    def _get_filetypes(self):
+        """
+        Get a dictionary filled with peewee rows referencing all of the filetypes in SQL db.
+
+        Returns
+        -------
+        dict
+            Key: String - name of filetype
+            Value: Model instance referencing row for filetype
+        """
+        return {ft: FileType.get(name=ft) for ft in FILETYPES}
+
+    def insert_files(self, files):
+        """
+        Insert files into File Table of SQL db. For them sets exists = True.
+
+        Files that already exist in db are ignored.
+
+        Parameters
+        ----------
+        files : List of String
+            list of filepaths
+        """
+        # pattern used to extract filetype of file from its name
+        pattern = re.compile(".*Z_(.*)_corr.*")
+
+        # represents each row in table
+        record = namedtuple(
+            "record", "filepath file_type_id, start_time end_time exists"
+        )
+
+        records = []  # list of `record`s
+
+        file_type_rows = self._get_filetypes()
+
+        print(f"File Tracker: Adding {len(files)} to database")
+
+        for f in files:
```

Comment:
You should remove already known files from this list before opening things. Otherwise you'll have to open 1000s of known files.